tests/operators/test_agnostic_aggregate_check.py

```diff
@@ -20,264 +20,204 @@
 Requires the unittest, pytest, and requests-mock Python libraries.
 
 """
-
 import logging
+import os
 import pathlib
+import random
 import unittest.mock
 
+import pytest
+from airflow.exceptions import BackfillUnfinished
 from airflow.hooks.sqlite_hook import SqliteHook
 from airflow.models import DAG, DagRun
 from airflow.providers.google.cloud.hooks.bigquery import BigQueryHook
 from airflow.providers.postgres.hooks.postgres import PostgresHook
 from airflow.utils import timezone
 
 import astro.sql as aql
+from astro.constants import SQLITE, SUPPORTED_DATABASES
 from astro.settings import SCHEMA
 from astro.sql.table import Table
+from tests.operators.utils import get_table_name, run_dag
 
 log = logging.getLogger(__name__)
 DEFAULT_DATE = timezone.datetime(2016, 1, 1)
+CWD = pathlib.Path(__file__).parent
+
+
+@pytest.fixture(scope="module")
+def table(request):
+    aggregate_table = Table(
+        "aggregate_check_test",
+        database="pagila",
+        conn_id="postgres_conn",
+        schema="airflow_test_dag",
+    )
+    aggregate_table_bigquery = Table(
+        "aggregate_check_test",
+        conn_id="bigquery",
+        schema=SCHEMA,
+    )
+    aggregate_table_sqlite = Table("aggregate_check_test", conn_id="sqlite_conn")
+    aggregate_table_snowflake = Table(
+        table_name=get_table_name("aggregate_check_test"),
+        database=os.getenv("SNOWFLAKE_DATABASE"),  # type: ignore
+        schema=os.getenv("SNOWFLAKE_SCHEMA"),  # type: ignore
+        warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),  # type: ignore
+        conn_id="snowflake_conn",
+    )
+    path = str(CWD) + "/../data/homes_merge_1.csv"
+    tables = {
+        "postgres": aggregate_table,
+        "bigquery": aggregate_table_bigquery,
+        "sqlite": aggregate_table_sqlite,
+        "snowflake": aggregate_table_snowflake,
+    }
+    aql.load_file(
+        path=path,
+        output_table=tables[request.param],
+    ).operator.execute({"run_id": "foo"})
+
+    yield tables[request.param]
```

Comment:
It can help readability to:
1. at the beginning of the fixture, have something like:
`database = request.param`
2. to explicitly use an if statement with three clauses in the fixture to highlight the fact the fixture does not aim to create three instances of table - but only one.